course/models.py

```diff
@@ -654,12 +654,16 @@ def get_queryset(self):
         return super().get_queryset().select_related(
             'course_instance', 'course_instance__course')
 
-    def get_visible(self, user=None):
+    def get_visible(self, user=None, is_course_staff=False):
```

Comment:
The `get_visible` method already has the user argument. A CourseModule always belongs to a CourseInstance (that defines assistants) and CourseInstance belongs to a Course (that defines teachers). The `is_course_staff` argument is not needed since the staff status can be queried from the database. The caller of the function already supplies the user and it should not be necessary to supply more info about the same user.

In addition, the CourseModuleManager is not tied to a single course. It queries all CourseModules from the database. The user is not a teacher on all courses. I think the `is_course_staff` argument makes the interface of this function feel very wrong. What do you think?

Reply:
I didn't really like this solution, but couldn't get any other way to work :)

Reply:
Did you try similar tricks that are used in the `get_course_visibility_filter` function here: https://github.com/apluslms/a-plus/blob/0cf8c3126d6011f08201764c3878b930ec4f3e31/course/models.py#L318